pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -497,9 +634,15 @@ func (r *MachineSyncReconciler) fetchCAPIInfraResources(ctx context.Context, cap
 
 // compareCAPIMachines compares CAPI machines a and b, and returns a list of differences, or none if there are none.
 func compareCAPIMachines(capiMachine1, capiMachine2 *capiv1beta1.Machine) []string {
+	capiMachine1Copy := capiMachine1.DeepCopy()
+	capiMachine2Copy := capiMachine2.DeepCopy()
+
+	capiMachine1Copy.OwnerReferences = nil
+	capiMachine2Copy.OwnerReferences = nil
+
```

Comment:
So if these are always nil, how do we pick up the diff there?

Reply:
This was incorrect.  I think I needed to compare the owner references differently if there could be extra ones that we would not be converting. This was not the way to do It and is not required as the current iteration only allows machine set ownerreference conversion.